### PR TITLE
Added fix for z-stack display in static stream.

### PR DESCRIPTION
**Pull request by amuskens**

### CODE REVIEW

File: src/odemis/acq/stream/_static.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def __init__(self, name, raw, *args, **kwargs):
         metadata = copy.copy(raw[0].metadata)
 
         # If there are 5 dims in CTZYX, eliminate CT and only take spatial dimensions
-        if raw[0].ndim > 3:
+        if raw[0].ndim >= 3:
             dims = metadata.get(model.MD_DIMS, "ZYX")
```
Comment (pieleric) [inline src/odemis/acq/stream/_static.py]: If you support more than 3 dims, you need to also check more carefully the default MD_DIMS. See the standard way in other code in odemis.

Comment (amuskens) [reply src/odemis/acq/stream/_static.py]: Any specific file to check for an example?

**Comment by pieleric** [comment]
An HDF5 fixed is now already merged. The static stream part of this PR looks fine. Could you drop the HDF5 commit from this PR, and squash the other two commits together?

**Comment by amuskens** [comment]
I think it is fixed now. Ready to merge?

File: src/odemis/acq/stream/_static.py [hunk 2]
```diff
@@ -177,8 +177,8 @@ def __init__(self, name, raw, *args, **kwargs):
         metadata = copy.copy(raw[0].metadata)
 
         # If there are 5 dims in CTZYX, eliminate CT and only take spatial dimensions
-        if raw[0].ndim > 3:
-            dims = metadata.get(model.MD_DIMS, "ZYX")
+        if raw[0].ndim >= 3:
+            dims = metadata.get(model.MD_DIMS, "CTZYX")
```
Comment (pieleric) [inline src/odemis/acq/stream/_static.py]: Please write the default as `"CTZYX"[-raw[0].ndim::]`.

File: src/odemis/acq/stream/_static.py [hunk 3]
```diff
@@ -189,10 +189,17 @@ def __init__(self, name, raw, *args, **kwargs):
                 if any(x > 1 for x in raw[0].shape[:1]):
                     logging.error("Higher dimensional data is being discarded.")
                 raw[0] = raw[0][0]
-            metadata[model.MD_DIMS] = "ZYX"
+
+            # Check if the Z dimension is empty
```
Comment (pieleric) [inline src/odemis/acq/stream/_static.py]: I'd suggest to write it this way, to make it clearer:
```
# Squash the Z dimension if it's empty
if  raw[0].shape[0] == 1:
    raw[0] = raw[0][0, :, :]
metadata[model.MD_DIMS] = "CTZYX"[-raw[0].ndim::]
```

**Pull request merged**